File: web-frontend/src/services/auth.service.ts
import { Me } from "../models/auth";
import { api } from "./axios.service";

export function saveSession(token: string) {
  return api.post<
    {
      oAuthToken: string;
    },
    void
  >("/auth/save-session", {
    oAuthToken: token,
  });
}

export async function autoLogin() {
  return api.post("auth/autologin");
}

export async function getCurrentAccount() {
  return api.get<Me>("auth/me").then((r) => r.data);
}

File: mobile/mobile/lib/services/auth-service.dart
import 'package:mobile/models/auth-models.dart';
import 'package:mobile/services/api-service.dart';

class AuthService {
  final ApiService _api = ApiService();

  Future<void> saveSession(String token) async {
    await _api.post(
      '/auth/save-session',
      data: {'oAuthToken': token},
    );
  }

  Future<void> autoLogin() async {
    await _api.post('/auth/autologin');
  }

  Future<Me> getCurrentAccount() async {
    final response = await _api.get('/auth/me');
    return Me.fromJson(response.data);
  }
}